modules/vNetPeering/2.0/deploy.json

```diff
@@ -0,0 +1,95 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "2.0.0.0",
+  "parameters": {
+    "spokeVnetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. The Name of the Virtual Network to add the peering to."
+      }
+    },
+    "peeringName": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. The Name of the virtual network peering resource."
+      }
+    },
+    "remoteVirtualNetworkId": {
+      "type": "string",
+      "metadata": {
+        "description": "Required. The Resource Id of the remote virtual network. The remove virtual network can be in the same or different region."
+      }
+    },
+    "allowVirtualNetworkAccess": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Optional. Whether the VMs in the local virtual network space would be able to access the VMs in remote virtual network space."
+      }
+    },
+    "allowForwardedTraffic": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Optional. Whether the forwarded traffic from the VMs in the local virtual network will be allowed/disallowed in remote virtual network."
+      }
+    },
+    "allowGatewayTransit": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Optional. If gateway links can be used in remote virtual networking to link to this virtual network."
+      }
+    },
+    "useRemoteGateways": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Optional. If remote gateways can be used on this virtual network. If the flag is set to true, and allowGatewayTransit on remote peering is also true, virtual network will use gateways of remote virtual network for transit. Only one peering can have this flag set to true. This flag cannot be set if virtual network already has a gateway."
+      }
+    }
+  },
+  "variables": {
+    "spokeToHubPeeringName": "[concat(parameters('spokeVnetName'),'/', parameters('peeringName'))]"
```

Comment:
Can this be changed to localToRemotePeeringName instead? There are exceptions where a user might need to connect two spokes (cost related reasons), so in this situation this name is not accurate. Because of this same reason, and since we have a remoteVirtualNetworkId, can we also change spokeVnetName to localVnetName instead, just to be more generic, thoughts?

Reply:
Yes that makes sense. I renamed the Parameter and Variable as requested.